railties/lib/rails/console/app.rb

```diff
@@ -34,5 +35,15 @@ def reload!(print = true)
       Rails.application.reloader.reload!
       true
     end
+
+    # reboot a brand new session and exit current context
+    def reboot!(print = true)
+      puts "Rebooting..." if print
+      existing_context = IRB.CurrentContext
+      Rails.application.reloader.reload!
+      IRB.start
```

Comment:
The console is not always IRB. It is configurable in `config.console`. We need to take in consideration this.